weather/www/js/app.js

```diff
@@ -0,0 +1,147 @@
+var app = angular.module('weather', ['ionic']);
+
+
+app.controller('HomeCtrl', function($scope, weatherService) {
+
+  
+
+  weatherService.get().then(function(response) {
+
+
+//got this idea from stack overflow but couldn't find the link
+    var monthNames = [
+      "January", "February", "March",
+      "April", "May", "June", "July",
+      "August", "September", "October",
+      "November", "December"
+    ];
+
+    var days = [
+      'Sunday', 'Monday', 'Tuesday', 'Wednesday',
+      'Thursday', 'Friday', 'Saturday'
+    ];
+
+
+
+    //w3school website
+    var currentTime = new Date();
+    var hour = currentTime.getHours();
+    var numDay = currentTime.getDate();
+    var day = currentTime.getDay();
+    var monthIndex = currentTime.getMonth();
+    var fullDate = (numDay + ", " + monthNames[monthIndex]);
+
+    console.log(day);
+
+    function time(x) {
+      var newTime = x + hour;
+      if (newTime > 12) {
+        newTime = newTime - 12;
+        if (newTime > 12) {
+          newTime = newTime - 12;
+          return (newTime) + "am";
+        } else {
+          return (newTime) + "pm";
+        }
+
+      } else {
+        return (newTime) + "am";
+      }
+
+    }
+    // got the idea from stack overflow: http://stackoverflow.com/questions/14657378/how-to-set-x-number-of-days-in-the-future-using-javascript
+
+    function dayOfTheWeek(x) {
+      var addDay = x + day;
+      if (addDay > 6) {
+        addDay = addDay - 6;
+      }
+      var newDay = days[addDay];
+      return (newDay);
+    }
+    console.log(dayOfTheWeek(6));
+
+
+    $scope.currentDate = fullDate;
+
+
+    $scope.weather = response.data;
+    console.log($scope.weather);
+    $scope.icon = $scope.weather.currently.icon;
+    $scope.currentTemp = Math.round($scope.weather.currently.temperature);
+    $scope.precipProbability = Math.round($scope.weather.currently.precipProbability);
+    $scope.humidity = Math.round($scope.weather.currently.humidity);
+    $scope.windSpeed = Math.round($scope.weather.currently.windSpeed);
+    $scope.summary = String($scope.weather.currently.summary);
+
+    $scope.hour1 = time(1) + ", " + Math.round($scope.weather.hourly.data[1].temperature) + "°F and " +
```

Comment:
Forecast.io gives you access to the timestamps at which the weather data was collected on.
So if I recall correctly, you should be able to do ``$scope.weather.hourly.data[1].time`` and get the time.
The time that forecast gives you is in *seconds from January 1, 1970* (UTC). To convert it to actually useful dates, see the Javascript *Date* documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
Note: Forecast.io gives you the time in seconds, but Javascript expects milliseconds, don't forget to convert!
**tldr;** You don't need the time(x) function and you don't need lines 77-115